install.sh: add support for Intel Meteor Lake binaries

kv2019i opened the pull request:
Add support for installing SOF firmare binary and related topology files according to the /lib/firmware layout expected by Linux SOF driver:

FW:
 - /lib/firmware/intel/sof-ipc4/mtl/sof-mtl.ri

Topology files:
 - /lib/firmware/intel/sof-ace-tplg/*.tplg

Reference:
 - https://github.com/torvalds/linux/blob/v6.5/sound/soc/sof/intel/pci-mtl.c#L23

kv2019i commented on install.sh:
```diff
@@ -28,22 +28,26 @@ main()
+    if test -e "$path/sof-$ver" -a -e "$path/sof-tplg-$ver" ; then
+        echo "SOF IPC3 SOF layout"
+    elif test -e "$path/sof-ipc4-$ver" -a -e "$path/sof-ace-tplg-$ver" ; then
+        echo "SOF IPC4 layout for Intel Meteor Lake (and newer)"
```
@marc-hb This is a multi-vendor and multi-version package and only one platfrom of one vendor is using the new layout at this point. If we drop other layouts, I think we should rename the install script name to "install-one-sof-bin-for-one-intel-platform.sh".

marc-hb replied:
We're talking across each other. Code speaks louder than words so is what I expected here:

```shell
for sdir in sof sof-ipc4 sof-ace-tplg sof-tplg;
```

In other words, same `for` iteration as below; as before.

marc-hb replied:
A "multi-version" tarball should NOT have multiple `$ver`, they should have all been merged in a single `sof-$ver` and `sof-tplg-$ver` directories by the `./tarball_multi_releases.bash` script.

When run without a tarball (NOT recommended), `./install.sh` supports a single `sof/` and a single `sof-tplg/` directory too.

Either way `./install.sh` should never know anything about "multi-version".

kv2019i commented:
V3 uploaded:
- rebased on top of multi-version-tarball changes
- updated README.md
- tested with v2.7-rc1 both installing from git working tree, and creating a release tarball and runing ./install.sh from the tarball

kv2019i commented:
V4 uploaded:
- actually include the README.md patchset
- fix a typo in install.sh patch

marc-hb commented on install.sh:
```diff
@@ -29,25 +29,30 @@ main()
-            set +x
-            die '%s already installed? (Re)move it first.\n' "${FW_DEST}/$sdir"
-        }
+
```
Please fix the comment above to mention symbolic links. That miss was not great  when the code that small but now there's a lot more code to read. Plus the `ln` line has never been very readable because of all the variables.

kv2019i replied:
@marc-hb I was wondering why you kept the intendation. Now fixed.

marc-hb replied:
I meant the source code comment when you're reading the code but not running it

kv2019i replied:
Aa, right. I think that is close enough. I agree we shouldn't expand this anymore, so if anything. I'd just remove the install exists checks (and make install.sh closer to the one-liner-run-rsync that is supposed to be in the end).

marc-hb commented on install.sh:
```diff
@@ -29,25 +29,30 @@ main()
-            die '%s already installed? (Re)move it first.\n' "${FW_DEST}/$sdir"
-        }
+
+    if test -e "$path/sof${optversuffix}" -a -e "$path/sof-tplg${optversuffix}" ; then
```
I kept the original and now wrong indentation to preserve `git blame` but now that all the code is changed there's no reason  not to indent correctly now.

kv2019i replied:
@marc-hb I was wondering why you kept the intendation. Now fixed.

marc-hb commented on install.sh:
```diff
@@ -29,25 +29,30 @@ main()
+    if test -e "$path/sof${optversuffix}" -a -e "$path/sof-tplg${optversuffix}" ; then
+        : # SOF IPC3 SOF layout
+    elif test -e "$path/sof-ipc4${optversuffix}" -a -e "$path/sof-ace-tplg${optversuffix}" ; then
+        : "SOF IPC4 layout for Intel Meteor Lake (and newer)"
```
Change this to a simple `printf` after making the `set -x` narrower

kv2019i replied:
Actually this doesn't print anything, that is just a no-op with a comment. The L44 was a mistake though, it should be a ": # comment"

marc-hb commented on install.sh:
```diff
@@ -29,25 +29,30 @@ main()
+
+    for sdir in sof sof-ipc4 sof-ace-tplg sof-tplg; do
+        if test -e "$path/$sdir${optversuffix}" ; then
+            ln -sT "$sdir-$ver" "${FW_DEST}/$sdir" || {
```
```suggestion
( set -x;  ln -sT "$sdir-$ver" "${FW_DEST}/$sdir" ) ||
   die '%s already installed? (Re)move it first.\n' "${FW_DEST}/$sdir"
```

kv2019i replied:
Done, thanks.

marc-hb reviewed (changes requested):
OK let's go with this. A good few style changes requested sorry, nothing major

kv2019i commented:
V5 uploaded:
- addressed everything else but the error print... not sure what is wrong in the error
